### DISCUSSION TITLE
BRouter parameters

### DISCUSSION THREAD

**Discussion by airald** [devemux86/cruiser#76] [Q&A] [answered]
I ride usually for many kind of roads. Some of them are not included in car or motorcycle profiles. If I use a bicycle profile, then BRouter priorizes paths instead of roads. So I'm trying to modify a profile in order to include this unpaved roads. Just that, not change the priority, just allow them to be included as available.

I'm searching about tech info for BRouter with not so much success. I've found a github site but car profiles are updated 7 years ago...

Which profile do you use? If you know if there's a group about that, please share it.

**Comment by devemux86** [comment, marked as answer]
- BRouter profiles website: https://brouter.de/brouter/costfunctions.html
- BRouter profiles list: https://brouter.de/brouter/profiles2/
- BRouter profiles guide: https://brouter.de/brouter/profile_developers_guide.txt
- BRouter profiles wiki: https://github.com/poutnikl/Brouter-profiles/wiki

**Comment by devemux86** [comment]
Default selection in Cruiser:

- Car | Motorcycle
  - Fastest -> car-fast
  - Shortest -> car-eco
- Bike
  - Fastest -> fastbike
  - Shortest -> trekking
- Walk -> shortest

**Comment by MotoUKRider** [comment]
Sorry, I can't help with an answer although I mostly got lucky with the modification I made to remove "gate" from the routing profile. I would suggest you open 2 profiles in Notepad++ and compare/contrast to see where the values/rules need to be added or changed.  Trial 'n' error plus a cup of your favourite drink would be the solution here!

This does raise the idea of having all those options (eg: avoidances and preferences) as checkboxes within the actual navigation app.

**Reply by airald** [reply]
Is what I'm doing (as usual) but at this moment, just errors. 

I also preffer an app to create custom profiles by simply changing checkboxes, assigning priorities, etc. BRouter have a very powerful algorithm. I'm sure it will fit my needs but more time is needed. I preffer that instead of "one button" apps.

**Reply by MotoUKRider** [reply]
It if helps, the changes I made for myself are in this recent thread: https://github.com/devemux86/cruiser/discussions/56  The last post I made shows a road feature that I wanted to fix.  There are some points to note about "open source" map editing, particularly attributes that affect routing of certain vehicles.

This wasn't an issue with Cruiser, of course - but *both* offline routing systems would not work as expected without some form of editing and a little coding skill.... The underlying issue here being that an average user would simply use another app.

**Comment by devemux86** [comment]
Questions about BRouter profiles can also be discussed with their authors in:
- [BRouter GitHub](https://github.com/abrensch/brouter/)
- [BRouter forum](https://groups.google.com/g/osm-android-bikerouting)

**Comment by devemux86** [comment]
> This does raise the idea of having all those options (eg: avoidances and preferences) as checkboxes within the actual navigation app.

> I also preffer an app to create custom profiles by simply changing checkboxes, assigning priorities, etc. BRouter have a very powerful algorithm.

This a very interesting idea and I am going to check what is possible in Cruiser. :+1:
